File: src/services/IServices/IRoleService.ts
import { Result } from "../../core/logic/Result";
import IRoleDTO from "../../dto/IRoleDTO";

export default interface IRoleService  {
  createRole(roleDTO: IRoleDTO): Promise<Result<IRoleDTO>>;
  updateRole(roleDTO: IRoleDTO): Promise<Result<IRoleDTO>>;

  getRoleById (roleId: string): Promise<Result<IRoleDTO>>;
  getRoles (): Promise<Result<Array<IRoleDTO>>>;
}

File: mpt/Services/IServices/IRoleService.cs
using Mpt.Core.Logic;
using Mpt.Domain.Roles;
using Mpt.Dtos;

namespace Mpt.IServices
{
    public interface IRoleService{

        Task<Result<RoleDto>> AddAsync(CreateRoleDto dto);
        Task<Result<RoleDto>> UpdateAsync(RoleDto dto);
        Task<Result<RoleDto>> GetByIdAsync(Guid id);
        Task<Result<List<RoleDto>>> GetAllAsync();
        Task<Result<RoleDto>> DeleteAsync(Guid id);
    }
}